From 4356e4fa5a5abb28dfc428397c53bf5b0f0efecf Mon Sep 17 00:00:00 2001
From: Xueling Li <xueling.li@amlogic.com>
Date: Fri, 16 Jul 2021 07:13:09 +0000
Subject: [PATCH 2/3] RDK: enable AVB in uboot  [1/1]

PD#SWPL-51970

Problem:
 dolbyVision need check firmware signature in uboot

Solution:
 enable AVB in uboot

Verify:
 check if system can boot up normally

Change-Id: I17b7be8c9a2e358926b477d4ac92d4e50b9ddd24
Signed-off-by: Xueling Li <xueling.li@amlogic.com>
---
 board/amlogic/configs/sc2_ah212.h            | 2 +-
 board/amlogic/defconfigs/sc2_ah212_defconfig | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/board/amlogic/configs/sc2_ah212.h b/board/amlogic/configs/sc2_ah212.h
index bc17c827e94..0328863823e 100644
--- a/board/amlogic/configs/sc2_ah212.h
+++ b/board/amlogic/configs/sc2_ah212.h
@@ -180,7 +180,7 @@
                 "if test ${active_slot} != normal; then "\
                     "setenv bootargs ${bootargs} androidboot.slot_suffix=${active_slot};"\
                 "fi;"\
-                "if test ${avb2} = 0; then "\
+                "if test ${avb2} = 1; then "\
                     "if test ${active_slot} = _a; then "\
                         "setenv bootargs ${bootargs} root=/dev/system_a;"\
                     "else if test ${active_slot} = _b; then "\
diff --git a/board/amlogic/defconfigs/sc2_ah212_defconfig b/board/amlogic/defconfigs/sc2_ah212_defconfig
index 1e5cc5cba99..b350dffb7b6 100644
--- a/board/amlogic/defconfigs/sc2_ah212_defconfig
+++ b/board/amlogic/defconfigs/sc2_ah212_defconfig
@@ -37,7 +37,7 @@ CONFIG_CMD_IR=y
 CONFIG_CMD_BMP=y
 CONFIG_CMD_CACHE=y
 # CONFIG_CMD_MISC is not set
-# CONFIG_CMD_BOOTCTOL_AVB=y
+CONFIG_CMD_BOOTCTOL_AVB=y
 CONFIG_CMD_WATCHDOG=y
 CONFIG_CMD_JTAG=y
 CONFIG_CMD_INI=y
-- 
2.31.1

